### PR TITLE
feat: link component change for rsp@v3

### PR DESCRIPTION
**Pull request by jianliao**
## Description
#553 

## How and where has this been tested?
 - **How this was tested:** 
 - **Browser(s) and OS(s) this was tested with:** 

## To-do list

- [x] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).

- [x] This pull request is ready to merge.

### CODE REVIEW

File: components/link/index.css [hunk 1]
```diff
@@ -17,9 +17,10 @@ governing permissions and limitations under the License.
   background-color: transparent;
   /* Remove gaps in links underline in iOS 8+ and Safari 8+. */
   -webkit-text-decoration-skip: objects;
-  text-decoration: none;
+  text-decoration: underline;
```
Comment (lazd) [inline components/link/index.css:20]: Not sure about this. I don't believe our links have an underline by default.

File: components/link/index.css [hunk 2]
```diff
@@ -17,9 +17,10 @@ governing permissions and limitations under the License.
...
   transition: color var(--spectrum-global-animation-duration-100) ease-in-out;
   outline: none;
+  cursor: pointer;
```
Comment (lazd) [inline components/link/index.css:23]: Hmm, I don't think this is necessary for `<a>`, but perhaps they're using the styles elsewhere, so it's fine.

File: components/link/index.css [hunk 3]
```diff
@@ -45,11 +46,14 @@ governing permissions and limitations under the License.
   }
 }
 
-.spectrum-Link--subtle,
-/** @deprecated */.spectrum-Link--quiet {
+.spectrum-Link--secondary {
   text-decoration: underline;
```
Comment (lazd) [inline components/link/index.css:50]: I think this is duplicative? We already have `text-decoration: underline` above

File: components/link/index.css [hunk 4]
```diff
@@ -45,11 +46,14 @@ governing permissions and limitations under the License.
   }
 }
 
-.spectrum-Link--subtle,
-/** @deprecated */.spectrum-Link--quiet {
+.spectrum-Link--secondary {
```
Comment (lazd) [inline components/link/index.css:49]: This needs to have `color: inherit` or something -- secondary links are supposed to look like the paragraph text. Not sure what to do on hover? https://spectrum.adobe.com/page/link/#Secondary-variant

File: components/link/metadata/link.yml [hunk 5]
```diff
@@ -1,34 +1,50 @@
+id: link-primary
 name: Link
 SpectrumSiteSlug: https://spectrum.adobe.com/page/link/
+sections:
+  - name: Migration Guide
+    description: |
+      ### Subtle variant
+      Subtle variant was removed. Please use Secondary.
```
Comment (lazd) [inline components/link/metadata/link.yml:8]: I believe `quiet` is the substitution for `subtle`, but I could be wrong.

File: components/link/metadata/link.yml [hunk 6]
```diff
@@ -1,34 +1,50 @@
...
       <div style="background-color: rgb(15, 121, 125); padding: 15px 20px; display: inline-block;">
         <p class="spectrum-Body" style="color: rgb(255, 255, 255);">
           Hello, this is my <a href="#" class="spectrum-Link spectrum-Link--overBackground"> spectrum-Link--overBackground</a>. This is just filler text, but if you keep reading maybe something good will happen.
         </p>
       </div>
+  - id: link-quiet-primary
+    name: Link - Quiet (Primary)
+    status: Verified
+    markup: |
+      <p> This uses <a href="#" class="spectrum-Link spectrum-Link--quiet">spectrum-Link--quiet</a>.</p>
```
Comment (lazd) [inline components/link/metadata/link.yml:30]: We don't have `primary` modifier, it seems links are primary by default. Is that what we want?

File: components/link/skin.css [hunk 7]
```diff
@@ -30,8 +30,7 @@ governing permissions and limitations under the License.
   }
 }
 
-.spectrum-Link--quiet,
-/** @deprecated */.spectrum-Link--subtle {
+.spectrum-Link--secondary{
   color: inherit;
```
Comment (lazd) [inline components/link/skin.css:34]: That looks right. Not sure what to do on hover though, is it correct as-is? 
https://spectrum.adobe.com/page/link/#Secondary-variant

File: components/link/index.css (hunk 3 above)
Comment (jianliao) [reply components/link/index.css]: I've cleaned up the duplicated `text-decoration` property.

File: components/link/metadata/link.yml (hunk 5 above)
Comment (jianliao) [reply components/link/metadata/link.yml]: Fixed.

File: components/link/metadata/link.yml (hunk 6 above)
Comment (jianliao) [reply components/link/metadata/link.yml:30]: @GarthDB, could you please confirm this? `primary` modifier got removed prior to this change.

File: components/link/skin.css (hunk 7 above)
Comment (jianliao) [reply components/link/skin.css:34]: hmm... This change was pulled in from RSP.

According to the description of Secondary variant on the spectrum-documentation site, the color should be [the same gray color as the paragraph text](https://spectrum.corp.adobe.com/page/link/#Secondary-variant).

Is it this reason that it used `inherit`? But I did find DNA tokens for  Secondary variant.

` --spectrum-link-secondary-text-color: var(--spectrum-alias-text-color);
   --spectrum-link-secondary-text-color-hover: var(--spectrum-alias-text-color-hover);
  --spectrum-link-secondary-text-color-key-focus: var(--spectrum-alias-text-color-hover);
`

**Comment by lazd** [comment]
The hover behavior seems to be missing, not sure what it's supposed to do? https://opensource.adobe.com/spectrum-css-vr-results/30854f8-result-2020-04-22_19-57-27/dist/docs/link.html

File: components/link/skin.css (hunk 7 above)
Comment (NateBaldwinDesign) [reply components/link/skin.css:34]: General rule of thumb we should try to abide by: Find DNA tokens and use them whenever available. For now, inheriting is displaying the correct colors, but if we change this in the future for *secondary links only* (no other type), then the change will not be picked up via DNA upgrade as expected.

**Comment by lazd** [comment]
@jianliao @GarthDB pushed a commit that drops the `.is-disabled` class and example -- this has been removed from the design system and now is a fine time to remove it.

**Comment by lazd** [comment]
AFAIK, @NateBaldwinDesign said there are no hover styles for Primary and Secondary links, so I believe this PR is good to go!

**Pull request merged**